File: go-client-sdk/pkg/models/operations/getrunstatus.go
package operations

import (
	"github.com/speakeasy-sdks/nolan-sdks/pkg/models/shared"
)

type GetRunStatusPathParams struct {
	ProjectID string `pathParam:"style=simple,explode=false,name=projectId"`
	RunID     string `pathParam:"style=simple,explode=false,name=runId"`
}

type GetRunStatusRequest struct {
	PathParams GetRunStatusPathParams
}

type GetRunStatusResponse struct {
	ContentType                  string
	ProjectStatusResponsePayload *shared.ProjectStatusResponsePayload
	StatusCode                   int
	TsoaErrorResponsePayload     *shared.TsoaErrorResponsePayload
}

File: typescript-client-sdk/src/sdk/models/operations/getrunstatus.ts
import { SpeakeasyBase, SpeakeasyMetadata } from "../../../internal/utils";
import * as shared from "../shared";
import { Type } from "class-transformer";


export class GetRunStatusPathParams extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "pathParam, style=simple;explode=false;name=projectId" })
  projectId: string;

  @SpeakeasyMetadata({ data: "pathParam, style=simple;explode=false;name=runId" })
  runId: string;
}

export class GetRunStatusRequest extends SpeakeasyBase {
  @SpeakeasyMetadata()
  pathParams: GetRunStatusPathParams;
}

export class GetRunStatusResponse extends SpeakeasyBase {
  @SpeakeasyMetadata()
  contentType: string;

  @SpeakeasyMetadata()
  projectStatusResponsePayload?: shared.ProjectStatusResponsePayload;

  @SpeakeasyMetadata()
  statusCode: number;

  @SpeakeasyMetadata()
  tsoaErrorResponsePayload?: shared.TsoaErrorResponsePayload;
}